packages/ripple-binary-codec/tools/generateDefinitions.js

```diff
@@ -0,0 +1,360 @@
+//  Ledger entry type processing
+////////////////////////////////////////////////////////////////////////
+addLine('  "LEDGER_ENTRY_TYPES": {')
+addLine('    "Any": -3,')
+addLine('    "Child": -2,')
+addLine('    "Invalid": -1,')
+
+const unhex = (x) => {
+  x = ('' + x).trim()
+  if (x.substr(0, 2) == '0x') return '' + parseInt(x)
+  if (x.substr(0, 1) == "'" && x.length == 3) return x.charCodeAt(1)
+  return x
+}
+hits = [
+  ...ledgerFormatsMacroFile.matchAll(
+    /^ *LEDGER_ENTRY\(lt[A-Z_]+ *, *([x0-9a-f]+) *, *([^,]+), \({$/gm,
+  ),
+]
+for (let x = 0; x < hits.length; ++x)
+  addLine(
+    '    "' +
+      hits[x][2] +
+      '": ' +
+      unhex(hits[x][1]) +
+      (x < hits.length - 1 ? ',' : ''),
+  )
+addLine('  },')
+
+////////////////////////////////////////////////////////////////////////
+//  SField processing
+////////////////////////////////////////////////////////////////////////
+addLine('  "FIELDS": [')
+// The ones that are harder to parse directly from SField.cpp
+addLine(`    [
+      "Generic",
+      {
+        "nth": 0,
+        "isVLEncoded": false,
+        "isSerialized": false,
+        "isSigningField": false,
+        "type": "Unknown"
+      }
+    ],
+    [
+      "Invalid",
+      {
+        "nth": -1,
+        "isVLEncoded": false,
+        "isSerialized": false,
+        "isSigningField": false,
+        "type": "Unknown"
+      }
+    ],
+    [
+      "ObjectEndMarker",
+      {
+        "nth": 1,
+        "isVLEncoded": false,
+        "isSerialized": true,
+        "isSigningField": true,
+        "type": "STObject"
+      }
+    ],
+    [
+      "ArrayEndMarker",
+      {
+        "nth": 1,
+        "isVLEncoded": false,
+        "isSerialized": true,
+        "isSigningField": true,
+        "type": "STArray"
+      }
+    ],
+    [
+      "hash",
+      {
+        "nth": 257,
+        "isVLEncoded": false,
+        "isSerialized": false,
+        "isSigningField": false,
+        "type": "Hash256"
+      }
+    ],
+    [
+      "index",
+      {
+        "nth": 258,
+        "isVLEncoded": false,
+        "isSerialized": false,
+        "isSigningField": false,
+        "type": "Hash256"
+      }
+    ],
+    [
+      "taker_gets_funded",
+      {
+        "nth": 258,
+        "isVLEncoded": false,
+        "isSerialized": false,
+        "isSigningField": false,
+        "type": "Amount"
+      }
+    ],
+    [
+      "taker_pays_funded",
+      {
+        "nth": 259,
+        "isVLEncoded": false,
+        "isSerialized": false,
+        "isSigningField": false,
+        "type": "Amount"
+      }
+    ],`)
+
+const isVLEncoded = (t) => {
+  if (t == 'VL' || t == 'ACCOUNT' || t == 'VECTOR256') return 'true'
+  return 'false'
+}
+
+const isSerialized = (t) => {
+  if (
+    t == 'LEDGERENTRY' ||
+    t == 'TRANSACTION' ||
+    t == 'VALIDATION' ||
+    t == 'METADATA'
+  )
+    return 'false'
+  return 'true'
+}
+
+const isSigningField = (t, notSigningField) => {
```

Comment:
When is `isSerialized` different from `isSigningField` ? They appear to be very similar

Reply:
One example where they differ is `TxnSignature`.

Reply:
Hmm yes. To close the loop on this thought, the other three exceptions are `Signature`, `MasterSignature` and `Signers` fields. They have been described with `SField::notSigning` in the `sfields.macro` file.